server/network/aoprotocol.py

```diff
@@ -605,39 +512,137 @@ def net_cmd_ms(self, args):
-                     additive, effect)
+        other_settings = (other_folder, other_emote, other_offset, other_flip)
+        send_args = self._build_send_arguments(
+            packet_28, other_settings, msg, charid_pair)
 
         self.client.area.last_ic_message = send_args
         self.client.area.send_command('MS', *send_args)
         self.server.area_manager.send_remote_command(
             target_area, 'MS', *send_args)
 
-        self.client.area.send_owner_command('MS',
-                                            *send_args[:5],
-                                            '[' + self.client.area.abbreviation + ']' + msg,
-                                            *send_args[5:]
-                                            )
-
+        self.client.area.send_owner_command(
+            "MS", *send_args[:5], "[" + self.client.area.abbreviation + "]" + msg, *send_args[5:])
         self.client.area.set_next_msg_delay(len(msg))
-        database.log_ic(self.client, self.client.area, showname, msg)
+        database.log_ic(self.client, self.client.area, packet_28.showname, msg)
 
         if (self.client.area.is_recording):
             self.client.area.recorded_messages.append(args)
 
+    def _packet_settings_are_valid(self, packet: MS_28) -> bool:
+        if packet.msg_type not in VALID_MESSAGE_TYPES:
+            return False
+
+        if packet.anim_type not in VALID_ANIM_TYPES:
+            return False
+
+        if packet.cid != self.client.char_id:
+            return False
+
+        if packet.sfx_delay < 0:
+            return False
+
+        if packet.evidence < 0:
+            return False
+
+        if packet.ding not in VALID_DINGS:
+            return False
+
+        if packet.color not in VALID_COLORS:
+            return False
+        
+        objection_modifier_as_str = str(packet.objection_modifier)
+        if '4' in objection_modifier_as_str and "<and>" not in objection_modifier_as_str:
+            if not objection_modifier_as_str.isdigit():
+                return False
+                
+        # TODO: Maybe make 15 a configuration setting?
+        if len(packet.showname) > 15:
+            self.client.send_ooc("Your IC showname is way too long!")
+            return False
+
+        return True
+    
+    def _transform_text(self, packet_text: str) -> str:
+        msg = self.dezalgo(packet_text)[:256]
+        if self.client.shaken:
+            msg = self.client.shake_message(msg)
+        if self.client.disemvowel:
+            msg = self.client.disemvowel_message(msg)
+        return msg
+
+    @staticmethod
+    def _apply_nonint_pre_settings(packet: MS_28) -> MS_28:
+        if packet.nonint_pre == 1:
+            if packet.objection_modifier in (1, 2, 3, 4, 23):
+                if packet.anim_type == 1 or packet.anim_type == 2:
+                    packet.anim_type = 0
+                elif packet.anim_type == 6:
+                    packet.anim_type = 5
+        return packet
```

Comment:
I wouldn't have put this stuff in here, I would have made it part of the packet class in my opinion.

Reply:
I like that idea. I'll probably make it part of the `@setter`